src/checker/http_checker.rs

```diff
@@ -489,6 +533,238 @@ mod tests {
+            ),
+            // ("pinning-test", "https://pinning-test.badssl.com/", "The certificate was not trusted."), for some reason this succeeds
+        ];
+
+        for (name, url, expected_msg) in test_cases {
+            let config = CheckConfig {
+                url: url.to_string(),
+                ..Default::default()
+            };
+
+            let result = checker.check_url(&config, &tick, "us-west").await;
+
+            assert_eq!(result.status, CheckStatus::Failure, "Test case: {}", name);
+            assert_eq!(
+                result.request_info.and_then(|i| i.http_status_code),
+                None,
+                "Test case: {}",
+                name
+            );
+            assert_eq!(
+                result.status_reason.as_ref().map(|r| r.status_type),
+                Some(CheckStatusReasonType::SslError),
+                "Test case: {}",
+                name
+            );
+            assert_eq!(
+                result.status_reason.map(|r| r.description).unwrap(),
+                expected_msg,
+                "Test case: {}",
+                name
+            );
+        }
+
+        // Test various DH key errors
+        let dh_test_cases = vec![
+            ("dh512", "https://dh512.badssl.com/"),
+            ("dh1024", "https://dh1024.badssl.com/"),
+            ("dh-small-subgroup", "https://dh-small-subgroup.badssl.com/"),
+            ("dh-composite", "https://dh-composite.badssl.com/"),
+        ];
+
+        for (name, url) in dh_test_cases {
+            let config = CheckConfig {
+                url: url.to_string(),
+                ..Default::default()
+            };
+
+            let result = checker.check_url(&config, &tick, "us-west").await;
+
+            assert_eq!(result.status, CheckStatus::Failure, "Test case: {}", name);
+            assert_eq!(
+                result.request_info.and_then(|i| i.http_status_code),
+                None,
+                "Test case: {}",
+                name
+            );
+            assert_eq!(
+                result.status_reason.as_ref().map(|r| r.status_type),
+                Some(CheckStatusReasonType::SslError),
+                "Test case: {}",
+                name
+            );
+            // The exact error message may vary by platform, so just verify it contains SSL/TLS error
+            assert!(
+                result
+                    .status_reason
+                    .as_ref()
+                    .map(|r| r.description.clone())
+                    .unwrap()
+                    .contains("handshake failure"),
+                "{}, {:?}",
+                name,
+                result
+                    .status_reason
+                    .as_ref()
+                    .map(|r| r.description.clone())
+                    .unwrap()
+            );
+        }
+    }
+    
+
+    #[tokio::test]
+    // #[cfg(target_os = "linux")]
+    async fn test_ssl_errors_linux() {
```

Comment:
it is probably not prudent to merge this test in as is. https://github.com/chromium/badssl.com

i feel like it would take a while to figure out how to mock this correctly or self-host the bad-ssl website for tests, so maybe its best to just comment out?